pgml-sdks/pgml/src/queries.rs

```diff
@@ -207,31 +214,20 @@ FROM
   INNER JOIN %s documents ON documents.id = chunks.document_id 
   ORDER BY 
   comparison.score DESC 
-  LIMIT 
-  $4;
 "#;
 
 pub const VECTOR_SEARCH: &str = r#"
-WITH comparison AS (
-  SELECT 
-    chunk_id, 
-    1 - (
-      %s.embedding <=> $1::vector 
-    ) AS score 
-  FROM 
-    %s 
-) 
 SELECT 
-  comparison.score, 
+  1 - (embeddings.embedding <=> $1::vector) score,
   chunks.chunk, 
   documents.metadata 
 FROM 
-  comparison 
-  INNER JOIN %s chunks ON chunks.id = comparison.chunk_id 
-  INNER JOIN %s documents ON documents.id = chunks.document_id 
-  ORDER BY 
-  comparison.score DESC 
-  LIMIT 
+  %s embeddings
+INNER JOIN %s chunks ON chunks.id = embeddings.chunk_id 
+INNER JOIN %s documents ON documents.id = chunks.document_id 
+ORDER BY 
+  embeddings.embedding <=> $1::vector ASC 
```

Comment:
This uses the index but does it mean the same? The sort on the right sorts by highest scores first, while this scores by lowest scores first. In fact, won't this give the worst results by similarity?

Reply:
Yes it means the same. <=> is the cosine distance function. We want ones with the lowest distance. 1 - <=> is the cosine similarity. You want ones with high similarity which is the same as low distance